actions/notification_actions.jsx

```diff
@@ -58,11 +79,17 @@ export function sendDesktopNotification(post, msgProps) {
             return;
         }
 
-        let notifyLevel = member && member.notify_props ? member.notify_props.desktop : NotificationLevels.DEFAULT;
-        if (notifyLevel === NotificationLevels.DEFAULT) {
-            notifyLevel = user && user.notify_props ? user.notify_props.desktop : NotificationLevels.ALL;
+        // if collapsedThreads is enabled and the post received is a Reply
+        // and the thread is not a UserThread (AKA followed thread) do not notify
+        if (isCrtReply) {
+            const thread = await dispatch(getThread(user.id, channel.team_id, post.root_id, false));
+            if (thread.error) {
+                return;
+            }
         }
```

Comment:
This could probably be done better, right now when CRT is on it calls the API on each reply received.
This is done to make sure to get a notification only when the thread is followed.

*when the thread is not followed the API returns an error code.

Reply:
Good work @koox00 🙂 Have you checked if we are also receiving `post.is_following` for this?

Reply:
Thanks @anurag6713, unfortunately it's not, AFAICT we don't return that kind of info on single posts.
So maybe one solution would be to support that server side,
and another solution would be to hook on `thread_updated` event for CRT notifications and `posted` event for non CRT replies and root posts.

Reply:
The websocket event received here also isn't per-person, so I don't think we could include whether or not the recipient is following the thread on the event.

That said, is there not a way to use the existing data loaded by the app to prevent having to make this check for each reply. That seems rather heavy.

Reply:
Alternatively, maybe we could have the server treats replies to followed threads as mentions so the client doesn't need to do as much work here? That's probably a lot larger change though since I don't know how much the notification code is aware of CRT. It would also make this parallel the old reply mentions feature as well.

Reply:
4/5 though that we should try to avoid repeatedly calling `getThread` through. If we could get it so that we're only calling it on the first time a user receives a post in the thread, that seems acceptable to me though.

Reply:
@hmhealey `notification.go` is an intimidating module as it does lots of stuff! 
But for the rest of the notifications to work (email, push) with CRT, we would need to modify it, so your proposition makes a lot of sense, thanks!

Reply:
@hmhealey logic has been moved to server and `notification_actions` has been reverted, for the most part, as it was :tada:
related server PR https://github.com/mattermost/mattermost-server/pull/18088